### ISSUE TITLE
Numerous "invalid section group" errors when building JSC on Ubuntu 18.10

### ISSUE THREAD

**Issue by GarrettVD** [googleprojectzero/fuzzilli#45]
After applying [webkit.patch](https://github.com/googleprojectzero/fuzzilli/blob/8d36345b5363689025e04cb3970cb86cbc04eb6b/Targets/JavaScriptCore/webkit.patch) to the WebKit commit [d6913c0043553080fa46caa49041e0b137db38a1](https://github.com/WebKit/webkit/tree/d6913c0043553080fa46caa49041e0b137db38a1) and running [fuzzbuild.sh](https://github.com/googleprojectzero/fuzzilli/blob/8d36345b5363689025e04cb3970cb86cbc04eb6b/Targets/JavaScriptCore/fuzzbuild.sh), I receive a bunch of "invalid section group" errors from clang.

Any ideas?

Full JSC build log: https://gist.github.com/GarrettVD/8d397d1d125b3d99cb49d253a51aed1f

**Comment by amarekano** [comment]
I did run into the same error when building jsc with llvm > 6.0. I didn't get around to debugging what caused this error.

I resolved this by using the llvm package that comes with the default distribution repo (i.e. 6.0). Hope this helps.

**Comment by amarekano** [comment]
@GarrettVD I've recently had to build jsc with clang-8 and I've figured out the issue here. The jsc build script uses gold as the default linker and gold seems to have trouble playing nice with clang-7 and higher. To disable the use of gold you would need to add '-DUSE_LD_GOLD=OFF' to your cmakeargs list.

Bear in mind that once you disable gold, the build script will use the linker that the symlink /usr/bin/ld points at (On ubuntu 18.10, it would point to bfd). On my setup I have reconfigured /usr/bin/ld to point to lld-8.

Best,
Amar
